src/type/chain/Chain.js

```diff
@@ -130,11 +130,21 @@ export const createChainClass = /* #__PURE__ */ factory(name, dependencies, ({ o
    */
   function chainify (fn) {
     return function () {
-      const args = [this.value] // `this` will be the context of a Chain instance
+      // Here, `this` will be the context of a Chain instance
+      if (arguments.length === 0) {
+        return new Chain(fn(this.value))
+      }
+      const args = [this.value]
       for (let i = 0; i < arguments.length; i++) {
         args[i + 1] = arguments[i]
       }
-
+      if (typed.isTypedFunction(fn)) {
+        const sigObject = typed.resolve(fn, args)
+        if (sigObject.params.length === 1) {
```

Comment:
Is there no need to explicitly check if  `sigObject` has a rest parameter? 

Does this work for example for a function that has a rest parameter as second argument (like `"Matrix, ...number" or so)`? We don't have that right now but it would be neat to write this check as specific as possible.

Reply:
Right now typed-function only supports rest parameters in final position. So the only way a rest parameter could be broken across an actual first argument in the Chain and actual additional arguments in the call is if it is the only parameter. If there were more parameters, they would precede the rest parameter and the first of them would be matching the actual first argument in the Chain, meaning the rest parameter would be matching only arguments in the current call, which is OK. Thus, the only problematic case is if the signature being matched has only a single parameter (which must be a rest parameter, as it is matching multiple actual arguments, the one in the chain and at least one more from the current call, since we dispose of the case of no actual arguments in the current call earlier). Hence we need only check whether the matching signature has only a single parameter. I will put a comment to this effect and push a commit. I do not think there is any action item.

Reply:
Thanks for the clarification 👍